portfolio/src/test/java/com/google/sps/FindScheduleTest.java

```diff
@@ -0,0 +1,136 @@
+        new Task("Task 4", "Second task", 45, LOW_PRIORITY));
+    Collection<Task> actual = schedule.greedy(events, tasks, TIME_0900, TIME_1700);
+    List<Task> expected =
+        Arrays.asList(new Task("Task 3", "Second task", 20, LOW_PRIORITY, STRING_0900AM),
+            new Task("Task 1", "First task", 30, LOW_PRIORITY, "2020-06-25T09:20:00Z"),
+            new Task("Task 2", "Second task", 30, LOW_PRIORITY, "2020-06-25T09:50:00Z"),
+            new Task("Task 4", "Second task", 45, LOW_PRIORITY, "2020-06-25T10:20:00Z"));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void manyEventsAndTasksGreedy() {
+    Collection<CalendarEvent> events =
+        Arrays.asList(new CalendarEvent("Event 1", "2020-06-25T09:00:00Z", "2020-06-25T09:30:00Z"),
+            new CalendarEvent("Event 2", "2020-06-25T10:00:00Z", "2020-06-25T10:45:00Z"),
+            new CalendarEvent("Event 3", "2020-06-25T12:00:00Z", "2020-06-25T13:00:00Z"),
+            new CalendarEvent("Event 4", "2020-06-25T13:30:00Z", "2020-06-25T14:00:00Z"),
+            new CalendarEvent("Event 5", "2020-06-25T15:00:00Z", "2020-06-25T15:30:00Z"),
+            new CalendarEvent("Event 6", "2020-06-25T17:30:00Z", "2020-06-25T18:00:00Z"));
+    Collection<Task> tasks = Arrays.asList(new Task("Task 1", "First task", 15, LOW_PRIORITY),
+        new Task("Task 2", "Second task", 30, LOW_PRIORITY),
+        new Task("Task 3", "Third task", 45, LOW_PRIORITY),
+        new Task("Task 4", "Fourth task", 15, LOW_PRIORITY),
+        new Task("Task 5", "Fifth task", 15, LOW_PRIORITY),
+        new Task("Task 6", "Sixth task", 45, LOW_PRIORITY),
+        new Task("Task 7", "Seventh task", 30, LOW_PRIORITY),
+        new Task("Task 8", "Eigth task", 15, LOW_PRIORITY),
+        new Task("Task 9", "Ninth task", 30, LOW_PRIORITY));
+    Collection<Task> actual = schedule.greedy(events, tasks, TIME_0900, TIME_1800);
+
+    List<Task> expected =
+        Arrays.asList(new Task("Task 1", "First task", 15, LOW_PRIORITY, "2020-06-25T09:30:00Z"),
+            new Task("Task 4", "Fourth task", 15, LOW_PRIORITY, "2020-06-25T09:45:00Z"),
+            new Task("Task 5", "Fifth task", 15, LOW_PRIORITY, "2020-06-25T10:45:00Z"),
+            new Task("Task 8", "Eigth task", 15, LOW_PRIORITY, "2020-06-25T11:00:00Z"),
+            new Task("Task 2", "Second task", 30, LOW_PRIORITY, "2020-06-25T11:15:00Z"),
+            new Task("Task 7", "Seventh task", 30, LOW_PRIORITY, "2020-06-25T13:00:00Z"),
+            new Task("Task 9", "Ninth task", 30, LOW_PRIORITY, "2020-06-25T14:00:00Z"),
+            new Task("Task 3", "Third task", 45, LOW_PRIORITY, "2020-06-25T15:30:00Z"),
+            new Task("Task 6", "Sixth task", 45, LOW_PRIORITY, "2020-06-25T16:15:00Z"));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  // TODO(tomasalvarez): Test edge cases
+  // 1: Events out of the scheduling bounds
+  // 2: Scheduling time is more than 24 hours
+  // 3: ???
```

Comment:
Maybe we could do something like only half of the tasks can be scheduled because there isn't enough free space for the day?

Reply:
I added a new TODO for this case, thank you for the suggestion.